---
title: "All_sequential"
output:
  pdf_document: default
  html_document: default
---
```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
"false"<=FALSE
"true"<=TRUE

knitr::opts_current$set(cache = TRUE, warning = FALSE,message = FALSE, cache.lazy = FALSE)
```


```{r, message=FALSE}
# Package names
packages <- c("ggrepel", "pheatmap", "DOSE", "clusterProfiler", "ensembldb", "annotables", "apeglm", "RColorBrewer", "vsn", "DESeq2", "ggplot2", "reshape2", "gage", "gageData", "AnnotationDbi", "tidyr", "dplyr", "org.Mm.eg.db", "gplots", "GO.db", "GOstats", "AnnotationHub", "pathview", "plyr", "tidyverse", "purrr", "GSEABase", "igraph", "vissE", "patchwork", "msigdb", "tibble", "msigdbr","fgsea", "GEOquery","EnhancedVolcano")

# Install packages not yet installed
installed_packages <- packages %in% rownames(installed.packages())
if (any(installed_packages == FALSE)) {
  BiocManager::install(packages[!installed_packages])
}

# Packages loading
invisible(lapply(packages, library, character.only = TRUE))
```

```{r}
library(readxl)
countData <- read_excel("Gene_count_matrix_DEseq2_V4_NOT normalized.xlsx")
colnames(countData)=c("gene_id","WT0_1","WT0_2","WT0_3","WT4_1","WT4_2","WT4_3","WT8_1","WT8_2","WT8_3","WT24_1","WT24_2","WT24_3","KO0_1","KO0_2","KO0_3","KO4_1","KO4_2","KO4_3","KO8_1","KO8_2","KO8_3","KO24_1","KO24_2")
countData=countData[,c(1:4,14:16)]
countData=aggregate(. ~gene_id , data = countData, sum)
#countData <- tibble::column_to_rownames(countData, "simple")
countData=column_to_rownames(countData,'gene_id')
treatment=c("WT","WT","WT","KO","KO","KO")
colData=as_tibble(cbind(colnames(countData),treatment))
colnames(colData)[1]="samplename"
dds=DESeqDataSetFromMatrix(countData = countData, colData = colData, design = ~treatment)
dds=estimateSizeFactors(dds)
sizeFactors(dds)
dds=DESeq(dds)
dds =dds[rowSums(counts(dds))>50,]
sizeFactors(dds)
rld0=vst(dds)
plotPCA(rld, intgroup=c("treatment"))

#normalized_counts0=counts(dds, normalized=TRUE)

normalized_counts0=rld



res_table0=lfcShrink(dds, coef=2)
#res_table0=results(dds, contrast = c("treatment", "WT", "KO"))

padj.cutoff <- 0.05
lfc.cutoff <- 1.5

normalized_counts0=counts(dds, normalized=TRUE)


# 
# normalized_counts0 <- normalized_counts0 %>%
#   data.frame() %>%
#   rownames_to_column(var="gene") %>% 
#   as_tibble()
# 
# normalized_counts0$entrez=mapIds(org.Mm.eg.db,key,keys = normalized_counts0$gene,column = "ENTREZID", keytype = "SYMBOL", multiVals = "first")
# 
# normalized_counts0$gene=normalized_counts0$entrez
# 
# normalized_counts0=normalized_counts0[,1:7]
# 
# normalized_counts0=aggregate(. ~gene , data = normalized_counts0, sum)
# 
# normalized_counts0 <- normalized_counts0 %>%
#   data.frame() %>%
#  column_to_rownames(var = "gene")







res_table0_tb <- res_table0 %>%
  data.frame() %>%
  rownames_to_column(var="gene") %>% 
  as_tibble()




sig0 <- res_table0_tb[which(res_table0_tb[,6]<0.05),]


normalized_counts_c0=counts(dds, normalized=TRUE)


normalized_counts_c0 <- normalized_counts_c0 %>%
  data.frame() %>%
  rownames_to_column(var="gene") %>% 
  as_tibble()

norm_sig0 <- normalized_counts_c0[,] %>% 
              dplyr::filter(normalized_counts_c0$gene %in% sig0$gene) %>%
	           data.frame() %>%
	           column_to_rownames(var = "gene") 

```




```{r}

EnhancedVolcano(res_table0,lab = rownames(res_table0),x = 'log2FoldChange',y = 'pvalue', title = 'WT vs KO time 0', pCutoff = 5e-2, FCcutoff = 1.5, pointSize = 2.0,labSize = 4.0)


```


```{r}


pheatmap(norm_sig0,fontsize = 2,cluster_cols = TRUE)

```


```{r}
##Code by Alex Soupir https://github.com/ACSoupir/Bioinformatics_RNASeq/blob/master/Mouse_RNA_Seq_p53_genotoxic.md

detectGroups <- function (x){  # x are col names
  tem <- gsub("_[0-9]*$","",x) # Remove all numbers from end
  #tem = gsub("_Rep|_rep|_REP","",tem)
  tem <- gsub("_$","",tem); # remove "_" from end
  tem <- gsub("_Rep$","",tem); # remove "_Rep" from end
  tem <- gsub("_rep$","",tem); # remove "_rep" from end
  tem <- gsub("_REP$","",tem)  # remove "_REP" from end
  return( tem )
}
```


```{r, fig.height=50, fig.width=25}

hclust2 <- function(x, method="average", ...)  # average linkage in hierarchical clustering
  hclust(x, method=method, ...)

n=1000 # number of top genes by standard deviation

x = assay(rld0)
if(n>dim(x)[1]) n = dim(x)[1] # max	as data

x = x[order(apply(x,1,sd),decreasing=TRUE),]  # sort genes by standard deviation

x = x[1:n,]   # only keep the n genes

# this will cutoff very large values, which could skew the color 
x=as.matrix(x[1:n,])-apply(x[1:n,],1,mean)
cutoff = median(unlist(x)) + 4*sd (unlist(x)) 
x[x>cutoff] <- cutoff
cutoff = median(unlist(x)) - 4*sd (unlist(x)) 
x[x< cutoff] <- cutoff
	
groups = detectGroups(colnames(x) )
groups.colors = rainbow(length(unique(groups) ) )


	lmat = rbind(c(5,4),c(0,1),c(3,2))
	lwid = c(1.5,4)
	lhei = c(1,.2,4)


heatmap.2(x, distfun = dist2,hclustfun=hclust2,
	 col=greenred(75), density.info="none", trace="none", scale="none", keysize=.5
	,key=T, symkey=F
	,ColSideColors=groups.colors[ as.factor(groups)]
	,margins=c(50,25)
	,cexRow=3
	,srtCol=25
	,cexCol=4  # size of font for sample names
	,lmat = lmat, lwid = lwid, lhei = lhei
	)
```


```{r}

res_table0_tb=aggregate(. ~gene , data = res_table0_tb, sum)


res_table0_tb <- res_table0_tb %>%
  data.frame() %>%
  rownames_to_column(var="gene") %>%
  as_tibble()

res_table0_tb$entrez=mapIds(org.Mm.eg.db,key,keys = res_table0_tb $gene,column = "ENTREZID", keytype = "SYMBOL", multiVals = "first")

res_table0_tb $gene=res_table0_tb$entrez

res_table0_tb=res_table0_tb[,1:7]

res_table0_tb=aggregate(. ~gene , data = res_table0_tb, sum)

res_table0_tb <- res_table0_tb %>%
  data.frame() %>%
 column_to_rownames(var = "gene")


```

```{r error=TRUE}


idx = grcm38$symbol %in% rownames(res_table0)

ids <- grcm38[idx, ]

non_duplicates <- which(duplicated(ids$symbol) == FALSE)

ids <- ids[non_duplicates, ] 

res_ids <- inner_join(res_table0_tb, ids, by=c("gene"="symbol"))     

all_genes <- as.character(res_ids$entrez)

sig0 <- dplyr::filter(res_ids, padj < 0.05)

sig0_genes <- as.character(sig0$entrez)

ego <- enrichGO(gene = sig0_genes, 
                universe = all_genes,
                keyType = "ENTREZID",
                OrgDb = org.Mm.eg.db, 
                ont = "BP", 
                qvalueCutoff = 0.05, 
                readable = TRUE)

cluster_summary <- data.frame(ego)

dotplot(ego, showCategory=15)
```



```{r error=TRUE}
zero_foldchanges <- sig0$log2FoldChange

names(zero_foldchanges) <- sig0$gene


zero_foldchanges <- zero_foldchanges[which(abs(zero_foldchanges)>1.5)]

```


```{r error=TRUE}
kegg_mouse <- kegg.gsets(species = "mouse", id.type = "kegg")

kegg.gs <- kegg_mouse$kg.sets[kegg_mouse$sigmet.idx]


keggres = gage(zero_foldchanges, gsets=kegg.gs, same.dir=T)


head(keggres$greater) #Pathways that are up-regulated

head(keggres$less) #Pathways that are down-regulated

sel_up <- keggres$greater[, "q.val"] < 0.05 & !is.na(keggres$greater[, "q.val"])

path_ids_up <- rownames(keggres$greater)[sel_up]


sel_down <- keggres$greater[, "q.val"] < 0.05 & !is.na(keggres$less[, "q.val"])


path_ids_down <- rownames(keggres$less)[sel_down]
path_ids_down

keggresids = substr(path_ids_up, start=1, stop=8)
```




```{r error=TRUE}
res_entrez <- filter(res_ids, entrez != "NA")
res_entrez <- res_entrez[which(duplicated(res_entrez$entrez) == F), ]
zero_foldchanges <- res_entrez$log2FoldChange
names(zero_foldchanges) <- res_entrez$entrez
zero_foldchanges <- sort(zero_foldchanges, decreasing = TRUE)

```


```{r error=TRUE}

gseaKEGG <- gseKEGG(geneList = zero_foldchanges, 
              organism = "mmu", # supported organisms listed below
              minGSSize = 10, 
              pvalueCutoff = 0.05, # padj cutoff value
              verbose = FALSE, by = "fgsea")
gseaKEGG_results <- gseaKEGG@result
View(gseaKEGG_results)
```












----------












```{r}
library(readxl)
countData <- read_excel("Gene_count_matrix_DEseq2_V4_NOT normalized.xlsx")
colnames(countData)=c("gene_id","WT0_1","WT0_2","WT0_3","WT4_1","WT4_2","WT4_3","WT8_1","WT8_2","WT8_3","WT24_1","WT24_2","WT24_3","KO0_1","KO0_2","KO0_3","KO4_1","KO4_2","KO4_3","KO8_1","KO8_2","KO8_3","KO24_1","KO24_2")
countData=countData[,c(1, 4:6,17:19)]
countData=aggregate(. ~gene_id , data = countData, sum)
#countData <- tibble::column_to_rownames(countData, "simple")
countData=column_to_rownames(countData,'gene_id')
treatment=c("WT","WT","WT","KO","KO","KO")
colData=as_tibble(cbind(colnames(countData),treatment))
colnames(colData)[1]="samplename"
dds=DESeqDataSetFromMatrix(countData = countData, colData = colData, design = ~treatment)
dds=estimateSizeFactors(dds)
sizeFactors(dds)
dds=DESeq(dds)
dds =dds[rowSums(counts(dds))>50,]
sizeFactors(dds)
rld=vst(dds)
plotPCA(rld, intgroup=c("treatment"))

#normalized_counts4=counts(dds, normalized=TRUE)

normalized_counts4=rld



res_table4=lfcShrink(dds, coef=2)
#res_table4=results(dds, contrast = c("treatment", "WT", "KO"))

padj.cutoff <- 0.05
lfc.cutoff <- 2

normalized_counts4=counts(dds, normalized=TRUE)


# 
# normalized_counts4 <- normalized_counts4 %>%
#   data.frame() %>%
#   rownames_to_column(var="gene") %>% 
#   as_tibble()
# 
# normalized_counts4$entrez=mapIds(org.Mm.eg.db,key,keys = normalized_counts4$gene,column = "ENTREZID", keytype = "SYMBOL", multiVals = "first")
# 
# normalized_counts4$gene=normalized_counts4$entrez
# 
# normalized_counts4=normalized_counts4[,1:7]
# 
# normalized_counts4=aggregate(. ~gene , data = normalized_counts4, sum)
# 
# normalized_counts4 <- normalized_counts4 %>%
#   data.frame() %>%
#  column_to_rownames(var = "gene")







res_table4_tb <- res_table4 %>%
  data.frame() %>%
  rownames_to_column(var="gene") %>% 
  as_tibble()




sig4 <- res_table4_tb %>%
        dplyr::filter(padj < padj.cutoff & abs(log2FoldChange) > lfc.cutoff)



normalized_counts_c4=counts(dds, normalized=TRUE)


normalized_counts_c4 <- normalized_counts_c4 %>%
  data.frame() %>%
  rownames_to_column(var="gene") %>% 
  as_tibble()

norm_sig4 <- normalized_counts_c4[,] %>% 
              dplyr::filter(normalized_counts_c4$gene %in% sig4$gene) %>%
	           data.frame() %>%
	           column_to_rownames(var = "gene") 

```




```{r}

EnhancedVolcano(res_table4,lab = rownames(res_table4),x = 'log2FoldChange',y = 'pvalue', title = 'WT vs KO time 0', pCutoff = 5e-2, FCcutoff = 1.5, pointSize = 2.0,labSize = 4.0)


```


```{r}

rg <- redgreen(20)
pheatmap(norm_sig4,fontsize = 2,color = rg)

```



```{r error=TRUE}

idx = grcm38$symbol %in% rownames(res_table4)

ids <- grcm38[idx, ]

non_duplicates <- which(duplicated(ids$symbol) == FALSE)

ids <- ids[non_duplicates, ] 

res_ids <- inner_join(res_table4_tb, ids, by=c("gene"="symbol"))     

all_genes <- as.character(res_ids$entrez)

sig4 <- dplyr::filter(res_ids, padj < 0.05)

sig4_genes <- as.character(sig4$entrez)

ego <- enrichGO(gene = sig4_genes, 
                universe = all_genes,
                keyType = "ENTREZID",
                OrgDb = org.Mm.eg.db, 
                ont = "BP", 
                pAdjustMethod = "BH", 
                qvalueCutoff = 0.005, 
                readable = TRUE)

cluster_summary <- data.frame(ego)

dotplot(ego, showCategory=15)
```



```{r error=TRUE}
four_foldchanges <- sig4$log2FoldChange

names(four_foldchanges) <- sig4$gene

four_foldchanges <- ifelse(four_foldchanges > 1.5, 1.5, four_foldchanges)
four_foldchanges <- ifelse(four_foldchanges < -1.5, -1.5, four_foldchanges)


```


```{r error=TRUE}
kegg_mouse <- kegg.gsets(species = "mouse", id.type = "kegg")
names(kegg_mouse)

kegg.gs <- kegg_mouse$kg.sets[kegg_mouse$sigmet.idx]
head(kegg.gs)

keggres = gage(four_foldchanges, gsets=kegg.gs, same.dir=T)
names(keggres)

head(keggres$greater) #Pathways that are up-regulated

head(keggres$less) #Pathways that are down-regulated

sel_up <- keggres$greater[, "q.val"] < 0.05 & !is.na(keggres$greater[, "q.val"])

path_ids_up <- rownames(keggres$greater)[sel_up]


sel_down <- keggres$greater[, "q.val"] < 0.05 & !is.na(keggres$less[, "q.val"])


path_ids_down <- rownames(keggres$less)[sel_down]
path_ids_down

keggresids = substr(path_ids_up, start=1, stop=8)
```




```{r error=TRUE}
res_entrez <- filter(res_ids, entrez != "NA")
res_entrez <- res_entrez[which(duplicated(res_entrez$entrez) == F), ]
four_foldchanges <- res_entrez$log2FoldChange
names(four_foldchanges) <- res_entrez$entrez
four_foldchanges <- sort(four_foldchanges, decreasing = TRUE)

```


```{r error=TRUE}

gseaKEGG <- gseKEGG(geneList = four_foldchanges, 
              organism = "mmu", # supported organisms listed below
              minGSSize = 10, 
              pvalueCutoff = 0.05, # padj cutoff value
              verbose = FALSE, by = "fgsea")
gseaKEGG_results <- gseaKEGG@result
View(gseaKEGG_results)
```







------------------







```{r}
library(readxl)
countData <- read_excel("Gene_count_matrix_DEseq2_V4_NOT normalized.xlsx")
colnames(countData)=c("gene_id","WT0_1","WT0_2","WT0_3","WT4_1","WT4_2","WT4_3","WT8_1","WT8_2","WT8_3","WT24_1","WT24_2","WT24_3","KO0_1","KO0_2","KO0_3","KO4_1","KO4_2","KO4_3","KO8_1","KO8_2","KO8_3","KO24_1","KO24_2")
countData=countData[,c(1,8:10,20:22)]
countData=aggregate(. ~gene_id , data = countData, sum)
#countData <- tibble::column_to_rownames(countData, "simple")
countData=column_to_rownames(countData,'gene_id')
treatment=c("WT","WT","WT","KO","KO","KO")
colData=as_tibble(cbind(colnames(countData),treatment))
colnames(colData)[1]="samplename"
dds=DESeqDataSetFromMatrix(countData = countData, colData = colData, design = ~treatment)
dds=estimateSizeFactors(dds)
sizeFactors(dds)
dds=DESeq(dds)
dds =dds[rowSums(counts(dds))>50,]
sizeFactors(dds)
rld=vst(dds)
plotPCA(rld, intgroup=c("treatment"))

#normalized_counts8=counts(dds, normalized=TRUE)

normalized_counts8=rld



res_table8=lfcShrink(dds, coef=2)
#res_table8=results(dds, contrast = c("treatment", "WT", "KO"))

padj.cutoff <- 0.05
lfc.cutoff <- 2

normalized_counts8=counts(dds, normalized=TRUE)


# 
# normalized_counts8 <- normalized_counts8 %>%
#   data.frame() %>%
#   rownames_to_column(var="gene") %>% 
#   as_tibble()
# 
# normalized_counts8$entrez=mapIds(org.Mm.eg.db,key,keys = normalized_counts8$gene,column = "ENTREZID", keytype = "SYMBOL", multiVals = "first")
# 
# normalized_counts8$gene=normalized_counts8$entrez
# 
# normalized_counts8=normalized_counts8[,1:7]
# 
# normalized_counts8=aggregate(. ~gene , data = normalized_counts8, sum)
# 
# normalized_counts8 <- normalized_counts8 %>%
#   data.frame() %>%
#  column_to_rownames(var = "gene")







res_table8_tb <- res_table8 %>%
  data.frame() %>%
  rownames_to_column(var="gene") %>% 
  as_tibble()




sig8 <- res_table8_tb %>%
        dplyr::filter(padj < padj.cutoff & abs(log2FoldChange) > lfc.cutoff)



normalized_counts_c8=counts(dds, normalized=TRUE)


normalized_counts_c8 <- normalized_counts_c8 %>%
  data.frame() %>%
  rownames_to_column(var="gene") %>% 
  as_tibble()

norm_sig8 <- normalized_counts_c8[,] %>% 
              dplyr::filter(normalized_counts_c8$gene %in% sig8$gene) %>%
	           data.frame() %>%
	           column_to_rownames(var = "gene") 

```




```{r}

EnhancedVolcano(res_table8,lab = rownames(res_table8),x = 'log2FoldChange',y = 'pvalue', title = 'WT vs KO time 0', pCutoff = 5e-2, FCcutoff = 1.5, pointSize = 2.0,labSize = 4.0)


```


```{r}

rg <- redgreen(20)
pheatmap(norm_sig8,fontsize = 2,color = rg)

```



```{r error=TRUE}

idx = grcm38$symbol %in% rownames(res_table8)

ids <- grcm38[idx, ]

non_duplicates <- which(duplicated(ids$symbol) == FALSE)

ids <- ids[non_duplicates, ] 

res_ids <- inner_join(res_table8_tb, ids, by=c("gene"="symbol"))     

all_genes <- as.character(res_ids$entrez)

sig8 <- dplyr::filter(res_ids, padj < 0.05)

sig8_genes <- as.character(sig8$entrez)

ego <- enrichGO(gene = sig8_genes, 
                universe = all_genes,
                keyType = "ENTREZID",
                OrgDb = org.Mm.eg.db, 
                ont = "BP", 
                pAdjustMethod = "BH", 
                qvalueCutoff = 0.05, 
                readable = TRUE)

cluster_summary <- data.frame(ego)

dotplot(ego, showCategory=15)
```



```{r error=TRUE}
eight_foldchanges <- sig8$log2FoldChange

names(eight_foldchanges) <- sig8$gene

eight_foldchanges <- ifelse(eight_foldchanges > 1.5, 1.5, eight_foldchanges)
eight_foldchanges <- ifelse(eight_foldchanges < -1.5, -1.5, eight_foldchanges)


```


```{r error=TRUE}
kegg_mouse <- kegg.gsets(species = "mouse", id.type = "kegg")
names(kegg_mouse)

kegg.gs <- kegg_mouse$kg.sets[kegg_mouse$sigmet.idx]
head(kegg.gs)

keggres = gage(eight_foldchanges, gsets=kegg.gs, same.dir=T)
names(keggres)

head(keggres$greater) #Pathways that are up-regulated

head(keggres$less) #Pathways that are down-regulated

sel_up <- keggres$greater[, "q.val"] < 0.05 & !is.na(keggres$greater[, "q.val"])

path_ids_up <- rownames(keggres$greater)[sel_up]


sel_down <- keggres$greater[, "q.val"] < 0.05 & !is.na(keggres$less[, "q.val"])


path_ids_down <- rownames(keggres$less)[sel_down]
path_ids_down

keggresids = substr(path_ids_up, start=1, stop=8)
```




```{r error=TRUE}
res_entrez <- filter(res_ids, entrez != "NA")
res_entrez <- res_entrez[which(duplicated(res_entrez$entrez) == F), ]
eight_foldchanges <- res_entrez$log2FoldChange
names(eight_foldchanges) <- res_entrez$entrez
eight_foldchanges <- sort(eight_foldchanges, decreasing = TRUE)

```


```{r error=TRUE}

gseaKEGG <- gseKEGG(geneList = eight_foldchanges, 
              organism = "mmu", # supported organisms listed below
              minGSSize = 10, 
              pvalueCutoff = 0.05, # padj cutoff value
              verbose = FALSE, by = "fgsea")
gseaKEGG_results <- gseaKEGG@result
View(gseaKEGG_results)
```








----------









```{r}
library(readxl)
countData <- read_excel("Gene_count_matrix_DEseq2_V4_NOT normalized.xlsx")
colnames(countData)=c("gene_id","WT0_1","WT0_2","WT0_3","WT4_1","WT4_2","WT4_3","WT8_1","WT8_2","WT8_3","WT24_1","WT24_2","WT24_3","KO0_1","KO0_2","KO0_3","KO4_1","KO4_2","KO4_3","KO8_1","KO8_2","KO8_3","KO24_1","KO24_2")
countData=countData[,c(1,11:13,23:24)]
countData=aggregate(. ~gene_id , data = countData, sum)
#countData <- tibble::column_to_rownames(countData, "simple")
countData=column_to_rownames(countData,'gene_id')
treatment=c("WT","WT","WT","KO","KO")
colData=as_tibble(cbind(colnames(countData),treatment))
colnames(colData)[1]="samplename"
dds=DESeqDataSetFromMatrix(countData = countData, colData = colData, design = ~treatment)
dds=estimateSizeFactors(dds)
sizeFactors(dds)
dds=DESeq(dds)
dds =dds[rowSums(counts(dds))>50,]
sizeFactors(dds)
rld=vst(dds)
plotPCA(rld, intgroup=c("treatment"))

#normalized_counts24=counts(dds, normalized=TRUE)

normalized_counts24=rld



res_table24=lfcShrink(dds, coef=2)
#res_table24=results(dds, contrast = c("treatment", "WT", "KO"))

padj.cutoff <- 0.05
lfc.cutoff <- 2

normalized_counts24=counts(dds, normalized=TRUE)


# 
# normalized_counts24 <- normalized_counts24 %>%
#   data.frame() %>%
#   rownames_to_column(var="gene") %>% 
#   as_tibble()
# 
# normalized_counts24$entrez=mapIds(org.Mm.eg.db,key,keys = normalized_counts24$gene,column = "ENTREZID", keytype = "SYMBOL", multiVals = "first")
# 
# normalized_counts24$gene=normalized_counts24$entrez
# 
# normalized_counts24=normalized_counts24[,1:7]
# 
# normalized_counts24=aggregate(. ~gene , data = normalized_counts24, sum)
# 
# normalized_counts24 <- normalized_counts24 %>%
#   data.frame() %>%
#  column_to_rownames(var = "gene")







res_table24_tb <- res_table24 %>%
  data.frame() %>%
  rownames_to_column(var="gene") %>% 
  as_tibble()




sig24 <- res_table24_tb %>%
        dplyr::filter(padj < padj.cutoff & abs(log2FoldChange) > lfc.cutoff)



normalized_counts_c24=counts(dds, normalized=TRUE)


normalized_counts_c24 <- normalized_counts_c24 %>%
  data.frame() %>%
  rownames_to_column(var="gene") %>% 
  as_tibble()

norm_sig24 <- normalized_counts_c24[,] %>% 
              dplyr::filter(normalized_counts_c24$gene %in% sig24$gene) %>%
	           data.frame() %>%
	           column_to_rownames(var = "gene") 

```




```{r}

EnhancedVolcano(res_table24,lab = rownames(res_table24),x = 'log2FoldChange',y = 'pvalue', title = 'WT vs KO time 0', pCutoff = 5e-2, FCcutoff = 1.5, pointSize = 2.0,labSize = 4.0)


```


```{r}

rg <- redgreen(20)
pheatmap(norm_sig24,fontsize = 2,color = rg)

```



```{r error=TRUE}

idx = grcm38$symbol %in% rownames(res_table24)

ids <- grcm38[idx, ]

non_duplicates <- which(duplicated(ids$symbol) == FALSE)

ids <- ids[non_duplicates, ] 

res_ids <- inner_join(res_table24_tb, ids, by=c("gene"="symbol"))     

all_genes <- as.character(res_ids$entrez)

sig24 <- dplyr::filter(res_ids, padj < 0.05)

sig24_genes <- as.character(sig24$entrez)

ego <- enrichGO(gene = sig24_genes, 
                universe = all_genes,
                keyType = "ENTREZID",
                OrgDb = org.Mm.eg.db, 
                ont = "BP", 
                pAdjustMethod = "BH", 
                qvalueCutoff = 0.05, 
                readable = TRUE)

cluster_summary <- data.frame(ego)

dotplot(ego, showCategory=15)
```



```{r error=TRUE}
twentyfour_foldchanges <- sig24$log2FoldChange

names(twentyfour_foldchanges) <- sig24$gene

twentyfour_foldchanges <- ifelse(twentyfour_foldchanges > 1.5, 1.5, twentyfour_foldchanges)
twentyfour_foldchanges <- ifelse(twentyfour_foldchanges < -1.5, -1.5, twentyfour_foldchanges)


```


```{r error=TRUE}
kegg_mouse <- kegg.gsets(species = "mouse", id.type = "kegg")
names(kegg_mouse)

kegg.gs <- kegg_mouse$kg.sets[kegg_mouse$sigmet.idx]
head(kegg.gs)

keggres = gage(twentyfour_foldchanges, gsets=kegg.gs, same.dir=T)
names(keggres)

head(keggres$greater) #Pathways that are up-regulated

head(keggres$less) #Pathways that are down-regulated

sel_up <- keggres$greater[, "q.val"] < 0.05 & !is.na(keggres$greater[, "q.val"])

path_ids_up <- rownames(keggres$greater)[sel_up]


sel_down <- keggres$greater[, "q.val"] < 0.05 & !is.na(keggres$less[, "q.val"])


path_ids_down <- rownames(keggres$less)[sel_down]
path_ids_down

keggresids = substr(path_ids_up, start=1, stop=8)
```




```{r error=TRUE}
res_entrez <- filter(res_ids, entrez != "NA")
res_entrez <- res_entrez[which(duplicated(res_entrez$entrez) == F), ]
twentyfour_foldchanges <- res_entrez$log2FoldChange
names(twentyfour_foldchanges) <- res_entrez$entrez
twentyfour_foldchanges <- sort(twentyfour_foldchanges, decreasing = TRUE)

```


```{r error=TRUE}

gseaKEGG <- gseKEGG(geneList = twentyfour_foldchanges, 
              organism = "mmu", # supported organisms listed below
              minGSSize = 10, 
              pvalueCutoff = 0.05, # padj cutoff value
              verbose = FALSE, by = "fgsea")
gseaKEGG_results <- gseaKEGG@result
View(gseaKEGG_results)
```






----------








```{r}
library(ggvenn)

ggvenn(list("24"=sig24$gene, "8"=sig8$gene,"4"=sig4$gene))

three_cond=intersect(intersect(sig24$gene,sig8$gene),sig4$gene)
three_cond
eight_n_four=intersect(sig4$gene,sig8$gene)
eight_n_four
eight_n_twentyfour=intersect(sig8$gene,sig24$gene)
eight_n_twentyfour
four_n_twentyfour=intersect(sig4$gene,sig24$gene)
four_n_twentyfour
```






```{r}
# 
# library(readxl)
# countData <- read_excel("Gene_count_matrix_DEseq2_V4_NOT normalized.xlsx")
# colnames(countData)=c("gene_id","WT0_1","WT0_2","WT0_3","WT4_1","WT4_2","WT4_3","WT8_1","WT8_2","WT8_3","WT24_1","WT24_2","WT24_3","KO0_1","KO0_2","KO0_3","KO4_1","KO4_2","KO4_3","KO8_1","KO8_2","KO8_3","KO24_1","KO24_2")
# countData=countData[,c(1,5:13,17:24)]
# countData=aggregate(. ~gene_id , data = countData, sum)
# #countData <- tibble::column_to_rownames(countData, "simple")
# countData=column_to_rownames(countData,'gene_id')
# treatment=c("WT","WT","WT","WT","WT","WT","WT","WT","WT","KO","KO","KO","KO","KO","KO","KO","KO")
# time=c(rep("4",3),rep("8",3),rep("24",3),rep("4",3),rep("8",3),rep("24",2))
# colData=as_tibble(cbind(colnames(countData),time,treatment))
# colData$group=str_c(treatment,time)
# colnames(colData)[1]="samplename"
# dds=DESeqDataSetFromMatrix(countData = countData, colData = colData, design = ~group)
# dds=estimateSizeFactors(dds)
# sizeFactors(dds)
# dds=DESeq(dds)
# dds =dds[rowSums(counts(dds))>50,]
# sizeFactors(dds)
# rld=vst(dds)
# plotPCA(rld, intgroup=c("treatment"))
# 
# #normalized_counts0=counts(dds, normalized=TRUE)
# 
# normalized_counts1=results(dds)
# 
# 
# 
# 
# 
# 
# 
# require(Biobase)
# normalized_counts1<-new("ExpressionSet", exprs=as.matrix(normalized_counts1))
# 
# 
# normalized_counts1$time= colData$group
# 
# 
# 
# 
# library(msigdbr)
# pathwaysDF <- msigdbr("mouse", category=c("H"))
# pathways <- split(as.character(pathwaysDF$entrez_gene), pathwaysDF$gs_name)
# 
# library(fgsea)
# set.seed(1)
# gesecaRes <- geseca(pathways, exprs(normalized_counts1), minSize = 50, maxSize = 1000, eps = 0)
# 
# 
# plotGesecaTable(gesecaRes |> head(20), pathways,E=exprs(normalized_counts1))
# 
# plotCoregulationProfile(pathway=pathways[["HALLMARK_GLYCOLYSIS"]],
#                         E=exprs(normalized_counts1), conditions=normalized_counts1$time)
# 
# 
# plotCoregulationProfile(pathway=pathways[["HALLMARK_INFLAMMATORY_RESPONSE"]],
#                         E=exprs(normalized_counts1), conditions=normalized_counts1$time)
# 
# 
# plotCoregulationProfile(pathway=pathways[["HALLMARK_FATTY_ACID_METABOLISM"]],
#                         E=exprs(normalized_counts1), conditions=normalized_counts1$time)
# 
# 
# plotCoregulationProfile(pathway=pathways[["HALLMARK_HYPOXIA"]],
#                         E=exprs(normalized_counts1), conditions=normalized_counts1$time)
# 
# 

```










```{r}
data("go.sets.mm")
data("go.subs.mm")
```


```{r}
go_bp_sets=go.sets.mm[go.subs.mm$BP]
go_bp_res=gage(exprs=zero_foldchanges, gsets=go.sets.mm, same.dir = TRUE)
```


```{r}
data("kegg.sets.mm")
data("sigmet.idx.mm")
kegg.sets.mm=kegg.sets.mm[sigmet.idx.mm]

```

```{r}
keggres=gage(exprs=zero_foldchanges, gsets = kegg.sets.mm, same.dir = TRUE)
keggrespathways_greater=data.frame(id=rownames(keggres$greater), keggres$greater) %>%
  tibble::as_tibble() %>%
  filter(row_number() <= 50) %>%
  .$id %>%
  as.character()
keggrespathways_greater
```

```{r}
keggrespathways_lower=data.frame(id=rownames(keggres$less),
  keggres$less) %>%
  tibble::as_tibble() %>%
  filter(row_number() <= 50) %>%
  .$id %>%
  as.character()
keggrespathways_lower
```

```{r}
keggresids=substr(keggrespathways_greater, start=1, stop=8)
keggresids
```


```{r}
##tmp=sapply(keggresids, function(pid) pathview(gene.data = foldchanges, pathway.id = pid, species = "mmu"))

```




